tests/convert/data_color_mine.rs

```diff
@@ -0,0 +1,245 @@
+    pub hsv_v: f32,
+}
+
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub struct ColorMine {
+    lab: Lab<f32>,
+    xyz: Xyz<f32>,
+    yxy: Yxy<f32>,
+    lch: Lch<f32>,
+    rgb: Rgb<f32>,
+    linear_rgb: Rgb<f32>,
+    hsl: Hsl<f32>,
+    hsv: Hsv<f32>,
+}
+
+fn gamma_to_linear(v: f32) -> f32 {
+    let eps = 0.04045_f32;
+    let a = 0.055_f32;
+    if v > eps {
+        ((v+a)/(1.0+a)).powf(2.4)
+    } else {
+        v / 12.92
+    }
+}
+
+// fn linear_to_gamma(v: f32) -> f32 {
+//     let eps = 0.0031308_f32;
+//     let a = 0.055_f32;
+//     if v > eps {
+//         (1.0+a)*v.powf(1.0/2.4) - a
+//     } else {
+//         v * 12.92
+//     }
+// }
+
+impl From<ColorMineRaw> for ColorMine {
+    fn from(src: ColorMineRaw) -> ColorMine {
+        ColorMine {
+            xyz: Xyz::new(src.xyz_x, src.xyz_y, src.xyz_z),
+            yxy: Yxy::new(src.yxy_x, src.yxy_y, src.yxy_luma),
+            lab: Lab::new(src.lab_l, src.lab_a, src.lab_b),
+            lch: Lch::new(src.lch_l, src.lch_c, src.lch_h.into()),
+            rgb: Rgb::new(src.rgb_r, src.rgb_g, src.rgb_b),
+            linear_rgb: Rgb::new(gamma_to_linear(src.rgb_r), gamma_to_linear(src.rgb_g), gamma_to_linear(src.rgb_b)),
+            hsl: Hsl::new(src.hsl_h.into(), src.hsl_s, src.hsl_l),
+            hsv: Hsv::new(src.hsv_h.into(), src.hsv_s, src.hsv_v),
+        }
+    }
+}
+
+macro_rules! impl_from_color {
+    ($self_ty:ident) => {
+        impl From<$self_ty<f32>> for ColorMine {
+            fn from(color: $self_ty<f32>) -> ColorMine {
+                ColorMine {
+                    lab: color.into_lab(),
+                    xyz: color.into_xyz(),
+                    yxy: color.into_yxy(),
+                    lch: color.into_lch(),
+                    linear_rgb: color.into_rgb(),
+                    rgb: color.into_rgb(),
+                    hsl: color.into_hsl(),
+                    hsv: color.into_hsv(),
+                }
+            }
+        }
+
+    }
+}
+
+impl_from_color!(Rgb);
+impl_from_color!(Xyz);
+impl_from_color!(Yxy);
+impl_from_color!(Lab);
+impl_from_color!(Lch);
+impl_from_color!(Hsl);
+impl_from_color!(Hsv);
+
+pub fn load_data(file_name: &str) -> Vec<ColorMine> {
+    let mut rdr = csv::Reader::from_file(file_name).expect("csv file could not be loaded in tests for color mine data");
+    let mut color_data: Vec<ColorMine> = Vec::new();
+    for record in rdr.decode() {
+        let r: ColorMineRaw = record.expect("color data could not be decoded in tests for color mine data");
+        color_data.push(r.into())
+    }
+    color_data
+}
+
+fn check_equal_cie(src: &ColorMine, tgt: &ColorMine) {
+    assert_color_eq!(src.xyz, tgt.xyz, [x,y,z]);
+    assert_color_eq!(src.yxy, tgt.yxy, [x,y,luma]);
+    assert_color_eq!(src.lab, tgt.lab, [l,a,b]);
+    assert_color_eq!(src.lch, tgt.lch, [l,chroma]);
+    // removed check due to issue with color mine huw conversion issues.
+    // assert_color_hue_eq!(src.lch, tgt.lch, [hue]);
```

Comment:
I've just tested this but yxy -> lch is failing with  check for difference of 0.1 and passes with difference of 4.0. The lab -> lch tests for red, blue and green pass

One failing test is for pink (from_yxy) where the difference is 3.1348 ( close to pi)
```
Expected value:  Lch { l: 0.8358, chroma: 0.19046874, hue: LabHue(7.82) }
Calculated value: Lch { l: 0.8358424, chroma: 0.19743656, hue: LabHue(10.954128) }
```

the lab values are off by a: 0.005 and b: 0.012.